### ISSUE TITLE
[BUG]: app crashed on ios when try to change video resource after it return from background

### ISSUE THREAD

**Issue by eslamvet** [TheWidlarzGroup/react-native-video#3900]
### Version

6.2.0

### What platforms are you having the problem on?

iOS

### System Version

Mac os senoma 14.3

### On what device are you experiencing the issue?

Real device, Simulator

### Architecture

Old architecture

### What happened?

this is the error message that appear in xcode 
Cannot remove an observer <NSKeyValueObservance 0x600000d07330> for the key path \"currentItem.videoComposition\" from <AVPlayer 0x6000000a64e0>, most likely because the value for the key \"currentItem\" has changed without an appropriate KVO notification being sent. Check the KVO-compliance of the AVPlayer class.

### Reproduction

the video player has play in background flag is playing video and went on background and then i open app and try to change video source so the app crash while doing that

**Comment by freeboub** [comment]
I tried to reproduce the issue with the sample app, I cannot reproduce it.
Can you please give more details on how to reproduce the issue ? (or maybe a sample app)

**Comment by eslamvet** [comment]
hi @freeboub 
just use Video Component with controls prop to true and playInBackground prop to false and  leave app in background the video will still playing in background then open app again and try to change video source the app will crash in ios

**Comment by eslamvet** [comment]
you can use this repo to reproduce the issue
https://github.com/ammar20305/Dummy
steps:
run app on ios simulator
then go to background
then open app again and click on button to change video source the app crash

**Comment by blazyliszek** [comment]
Perhaps this issue was accidentally fixed and reintroduced. I faced the exact same crash, but from my investigation the bug came to me with [this change](https://github.com/TheWidlarzGroup/react-native-video/pull/3973). (I knew the range of possibly faulty versions to try out from my bug reporting tools, so I tried out a couple by using different commit-hashes to narrow it down.) This change was released in `6.4.0`.
In general, this is where the crash happens:
https://github.com/TheWidlarzGroup/react-native-video/blob/38aa2b057a496f96e6bf176da984df074ded418f/ios/Video/RCTVideo.swift#L491
